CMakeLists.txt

```diff
@@ -0,0 +1,79 @@
+##===----------------------------------------------------------------------===##
+##
+## This source file is part of the Swift open source project
+##
+## Copyright (c) 2024 Apple Inc. and the Swift project authors
+## Licensed under Apache License v2.0
+##
+## See LICENSE.txt for license information
+## See CONTRIBUTORS.md for the list of Swift project authors
+##
+## SPDX-License-Identifier: Apache-2.0
+##
+##===----------------------------------------------------------------------===##
+
+cmake_minimum_required(VERSION 3.24)
+
+project(SwiftFoundationICU
+    LANGUAGES CXX Swift)
+
+set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake/modules)
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
+set(CMAKE_Swift_MODULE_DIRECTORY ${CMAKE_BINARY_DIR}/swift)
+
+option(BUILD_SHARED_LIBS ON)
```

Comment:
Don't need this bit. CMake knows about `BUILD_SHARED_LIBS` already and will implicitly set any `add_library` that doesn't specify `SHARED` or `STATIC` with the selection there.

https://cmake.org/cmake/help/latest/variable/BUILD_SHARED_LIBS.html

Reply:
I think that this is trying to build as shared libraries by default.

Reply:
Okay, then it's fine to leave this line. Don't need 30 - 34 though. Those are handled implicitly by `add_library` if the library type isn't specified.